kivy/core/window/__init__.py

```diff
@@ -629,11 +629,29 @@ def _get_android_kheight(self):
             import android
         return android.get_keyboard_height()
 
+    def _get_kivy_vkheight(self):
+        mode = Config.get('kivy', 'keyboard_mode')
+        smode = self.softinput_mode
+        if (
+            (mode == 'dock' or mode == 'systemanddock')
+            and self._vkeyboard_cls is not None
+        ):
+            for w in self.children:
+                if isinstance(w, VKeyboard):
+                    vkeyboard_height = w.height * w.scale
+                    if smode == 'pan':
+                        return vkeyboard_height
+                    elif smode == 'below_target':
+                        return vkeyboard_height - w.target.y
+        return 0
+
     def _get_kheight(self):
         if platform == 'android':
             return self._get_android_kheight()
-        if platform == 'ios':
+        elif platform == 'ios':
             return self._get_ios_kheight()
+        else:
```

Comment:
These three lines can merge into one: `return self._get_kivy_vkheight()`

Reply:
Replace the conditions?
I don't think I understand, could you be more specific?

Reply:
```
        else:
            return self._get_kivy_vkheight()
        return 0
```
replace it with: `return self._get_kivy_vkheight()`

Reply:
Understood!